app/experimenter/static/rapid/contexts/experiment/actions.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  ExperimentReducerActionType,
+  ExperimentData,
+} from "experimenter-types/experiment";
+import { ExperimentReducerAction } from "experimenter-types/experiment";
+
+export const fetchExperiment = (experimentSlug: string) => async (
+  experimentData: ExperimentData,
+  dispatch: React.Dispatch<ExperimentReducerAction>,
+): Promise<void> => {
+  const response = await fetch(`/api/v3/experiments/${experimentSlug}/`, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  const data = await response.json();
+  dispatch({
+    type: ExperimentReducerActionType.UPDATE_STATE,
+    state: data,
+  });
+};
+
+export const saveExperiment = async (
+  experimentSlug: string,
+  formData: ExperimentData,
+): Promise<Response> => {
+  const url = experimentSlug
+    ? `/api/v3/experiments/${experimentSlug}/`
+    : "/api/v3/experiments/";
+  return await fetch(url, {
+    method: experimentSlug ? "PUT" : "POST",
```

Comment:
i tried to return just the response here, but i realized that we need the slug from the response in the experiment form